nav2_safety_nodes/CMakeLists.txt

```diff
@@ -0,0 +1,112 @@
+cmake_minimum_required(VERSION 3.8)
+project(nav2_safety_nodes)
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_lifecycle REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(laser_geometry REQUIRED)
+find_package(nav2_util REQUIRED)
+find_package(nav2_common REQUIRED)
+find_package(nav2_msgs REQUIRED)
+find_package(visualization_msgs REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+find_package(tf2_ros REQUIRED)
+find_package(tf2_sensor_msgs REQUIRED)
+
+nav2_package()
+
+link_libraries(stdc++fs)
+
+include_directories(
+  include
+)
+# prevent pluginlib from using boost
+# target_compile_definitions(nav2_safety_nodes_core PUBLIC "PLUGINLIB__DISABLE_BOOST_FUNCTIONS")
+
+add_library(nav2_safety_node_core SHARED
+  src/nav2_safety_node.cpp
+)
+
+set(dependencies
+  rclcpp
+  rclcpp_lifecycle
+  std_msgs
+  sensor_msgs
+  geometry_msgs
+  visualization_msgs
+  laser_geometry
+  nav2_util
+  pluginlib
+  tf2
+  tf2_geometry_msgs
+  tf2_ros
+  tf2_sensor_msgs
+)
+
+ament_target_dependencies(nav2_safety_node_core
+  ${dependencies}
+)
+
+add_executable(nav2_safety_node
+  src/main.cpp
+)
+
+target_link_libraries(nav2_safety_node
+  nav2_safety_node_core
+)
+
```

Comment:
1 space between each thing -- apply for the full file. Lots of irregular spacing below